core/arch/arm/kernel/thread.c

```diff
@@ -613,6 +614,31 @@ size_t thread_stack_size(void)
 	return STACK_THREAD_SIZE;
 }
 
+void get_stack_limits(vaddr_t *start, vaddr_t *end)
+{
+	uint32_t exceptions = thread_mask_exceptions(THREAD_EXCP_FOREIGN_INTR);
+	struct thread_core_local *l = thread_get_core_local();
```

Comment:
I'd prefer if we split out a helper function `static struct thread_core_local *get_core_local(unsigned int pos)` from
`thread_get_core_local()` and use that helper function here, we wan't the asserts but not yet another call to `get_core_pos()`.

Reply:
OK